src/Consolonia.GuiCS/WindowsDriver.cs

```diff
@@ -2,294 +2,63 @@
-			public short Y;
-
-			public Coord (short X, short Y)
-			{
-				this.X = X;
-				this.Y = Y;
-			}
-			public override string ToString () => $"({X},{Y})";
-		};
-
-		[StructLayout (LayoutKind.Explicit, CharSet = CharSet.Unicode)]
-		public struct CharUnion {
-			[FieldOffset (0)] public char UnicodeChar;
-			[FieldOffset (0)] public byte AsciiChar;
-		}
-
-		[StructLayout (LayoutKind.Sequential)]
-		public struct SmallRect {
-			public short Left;
-			public short Top;
-			public short Right;
-			public short Bottom;
-
-			public SmallRect (short left, short top, short right, short bottom)
-			{
-				Left = left;
-				Top = top;
-				Right = right;
-				Bottom = bottom;
-			}
-
-
-			public override string ToString ()
-			{
-				return $"Left={Left},Top={Top},Right={Right},Bottom={Bottom}";
-			}
-		}
-
-		[DllImport ("kernel32.dll", SetLastError = true)]
-		static extern IntPtr GetStdHandle (int nStdHandle);
-
-		[DllImport ("kernel32.dll", EntryPoint = "ReadConsoleInputW", CharSet = CharSet.Unicode)]
-		public static extern bool ReadConsoleInput (
-			IntPtr hConsoleInput,
-			IntPtr lpBuffer,
-			uint nLength,
-			out uint lpNumberOfEventsRead);
-
-	
-		[StructLayout (LayoutKind.Sequential)]
-		public struct ConsoleCursorInfo {
-			public uint dwSize;
-			public bool bVisible;
-		}
-
-		[DllImport ("kernel32.dll")]
-		static extern bool GetConsoleMode (IntPtr hConsoleHandle, out uint lpMode);
-
-
-		[DllImport ("kernel32.dll")]
-		static extern bool SetConsoleMode (IntPtr hConsoleHandle, uint dwMode);
-
-		public InputRecord [] ReadConsoleInput ()
-		{
-			const int bufferSize = 1;
-			var pRecord = Marshal.AllocHGlobal (Marshal.SizeOf<InputRecord> () * bufferSize);
-			try {
-				ReadConsoleInput (InputHandle, pRecord, bufferSize,
-					out var numberEventsRead);
-
-				return numberEventsRead == 0
-					? null
-					: new [] { Marshal.PtrToStructure<InputRecord> (pRecord) };
-			} catch (Exception) {
-				return null;
-			} finally {
-				Marshal.FreeHGlobal (pRecord);
-			}
-		}
-
-		[StructLayout (LayoutKind.Sequential)]
-		public struct CONSOLE_SCREEN_BUFFER_INFOEX {
-			public uint cbSize;
-			public Coord dwSize;
-			public Coord dwCursorPosition;
-			public ushort wAttributes;
-			public SmallRect srWindow;
-			public Coord dwMaximumWindowSize;
-			public ushort wPopupAttributes;
-			public bool bFullscreenSupported;
-
-			[MarshalAs (UnmanagedType.ByValArray, SizeConst = 16)]
-			public COLORREF [] ColorTable;
-		}
-
-		[StructLayout (LayoutKind.Explicit, Size = 4)]
-		public struct COLORREF {
-			public COLORREF (byte r, byte g, byte b)
-			{
-				Value = 0;
-				R = r;
-				G = g;
-				B = b;
-			}
-
-			public COLORREF (uint value)
-			{
-				R = 0;
-				G = 0;
-				B = 0;
-				Value = value & 0x00FFFFFF;
-			}
-
-			[FieldOffset (0)]
-			public byte R;
-			[FieldOffset (1)]
-			public byte G;
-			[FieldOffset (2)]
-			public byte B;
-
-			[FieldOffset (0)]
-			public uint Value;
-		}
+    [SupportedOSPlatform("windows")]
```

Comment:
Just curios what does it do?

Reply:
If the class is used in code which can be accessed by cross platform it will give you a warning/error at build time. You then can disable the warning if you know it's ok. For example:
```csharp
if (Environment.Windows)
    console = new Win32Console(...);
```